internals/secrethub/init.go

```diff
@@ -0,0 +1,178 @@
+package secrethub
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/progress"
+	"github.com/secrethub/secrethub-cli/internals/cli/ui"
+	"github.com/secrethub/secrethub-cli/internals/secrethub/command"
+	"github.com/secrethub/secrethub-go/internals/errio"
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/credentials"
+)
+
+// InitCommand configures the user's SecretHub account for use on this machine.
+type InitCommand struct {
+	backupCode                  string
+	force                       bool
+	io                          ui.IO
+	newClient                   newClientFunc
+	newClientWithoutCredentials func(credentials.Provider) (secrethub.ClientInterface, error)
+	credentialStore             CredentialConfig
+	progressPrinter             progress.Printer
+}
+
+// NewInitCommand creates a new InitCommand.
+func NewInitCommand(io ui.IO, newClient newClientFunc, newClientWithoutCredentials func(credentials.Provider) (secrethub.ClientInterface, error), credentialStore CredentialConfig) *InitCommand {
+	return &InitCommand{
+		io:                          io,
+		newClient:                   newClient,
+		newClientWithoutCredentials: newClientWithoutCredentials,
+		credentialStore:             credentialStore,
+		progressPrinter:             progress.NewPrinter(io.Stdout(), 500*time.Millisecond),
+	}
+}
+
+// Register registers the command, arguments and flags on the provided Registerer.
+func (cmd *InitCommand) Register(r command.Registerer) {
+	clause := r.Command("init", "Initialize the SecretHub client for first use on this device.")
+	clause.Flag("backup-code", "The backup code used for initializing the account on this device.").StringVar(&cmd.backupCode)
```

Comment:
Maybe add something with `restore` or `existing` to make extra clear when this is flag is used.

e.g. `Backup code used to restore an existing account to this device.`

Reply:
Done.